### PR TITLE
DMP-2466-Retention-Policies-Primary-Page

### PR DESCRIPTION
**Pull request by bpires2**
### Jira link (if applicable)

[DMP-2466](https://tools.hmcts.net/jira/browse/DMP-2466)

### Change description ###

- Add Retention policies primary page

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/app/admin/models/retention-policies/retention-policy-types-data.interface.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface RetentionPolicyTypesData {
```
Comment (deanbullock-hmcts) [inline src/app/admin/models/retention-policies/retention-policy-types-data.interface.ts:1]: I would drop 'Types' from the name

File: src/app/admin/models/retention-policies/retention-policy-types.type.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { DateTime } from 'luxon';
+
+export type RetentionPolicyTypes = {
```
Comment (deanbullock-hmcts) [inline src/app/admin/models/retention-policies/retention-policy-types.type.ts:3]: Same here, 'Types' seems redundant?

File: darts-api-stub/admin/retention-policies/retention-policies.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    display_name: 'Legacy Standard',
+    description: 'lorem ipsum',
+    fixed_policy_key: 2,
+    duration: '1Y6M0D',
+    policy_start_at: '2024-01-01T00:00:00Z',
+    policy_end_at: '2099-01-01T00:00:00Z',
+  },
+  {
+    id: 2,
+    name: 'DARTS Standard Retention v3',
```
Comment (deanbullock-hmcts) [inline darts-api-stub/admin/retention-policies/retention-policies.js:29]: Perhaps vary the stub data a little so it doesn't all look the same in the UI?

File: cypress/e2e/functional/admin-portal/retention-policies-spec.cy.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    cy.get('#inactive-policies-table').contains('td', '31 Jan 2024 12:00 AM');
+  });
+});
+
+// cy.get('#annotationsTable')
+//     .contains(fileName)
+//     .parent('tr')
+//     .then((row) => {
+//         cy.wrap(row).find('td').contains('Download').click();
+//     });
```
Comment (deanbullock-hmcts) [inline cypress/e2e/functional/admin-portal/retention-policies-spec.cy.js:38]: What's this commented code for?

Comment (bpires2) [reply cypress/e2e/functional/admin-portal/retention-policies-spec.cy.js:38]: I used this as a standard template for my functional table tests, Deleted now

File: darts-api-stub/admin/retention-policies/retention-policies.js (hunk 3 above)
Comment (bpires2) [reply darts-api-stub/admin/retention-policies/retention-policies.js:29]: Changed now

File: src/app/admin/models/retention-policies/retention-policy-types.type.ts (hunk 2 above)
Comment (bpires2) [reply src/app/admin/models/retention-policies/retention-policy-types.type.ts:3]: Renamed

File: src/app/admin/models/retention-policies/retention-policy-types-data.interface.ts (hunk 1 above)
Comment (bpires2) [reply src/app/admin/models/retention-policies/retention-policy-types-data.interface.ts:1]: Renamed

**Review by deanbullock-hmcts — APPROVED** [review]
🦒

**Pull request merged**